src/index.ts

```diff
@@ -85,7 +85,7 @@ export const swagger = async <Path extends string = '/swagger'>(
 		const scalarConfiguration: ReferenceConfiguration = {
 			spec: {
 				...scalarConfig.spec,
-				url: openAPISpecUrl
+				url: `${request.url}/json`
```

Comment:
This has 2 potential problems:
- if the swagger UI url includes trailing slash, the spec url will contain `//json`
- in some cases (e.g. behind reverse proxies), the hostname as well as the protocol is not actually correct in `request.url` and you can only rely on `pathname` being correct.
```suggestion
				url: `${new URL(request.url).pathname.replace(/\/$/, "")}/json`
```